.github/workflows/release.yml

```diff
@@ -1,59 +1,174 @@
-          GOARCH: ${{ matrix.GOARCH }}
-          CGO_ENABLED: 0
+      - name: Checkout codebase
+        uses: actions/checkout@v2
+
+      - name: Show workflow information 
+        id: get_filename
         run: |
-          go version
-          go build -tags netgo -ldflags "-X github.com/dutchcoders/transfer.sh/cmd.Version=${GITHUB_REF##*/} -a -s -w -extldflags '-static'" -o ./artifacts/transfersh-${GITHUB_REF##*/}-${{ matrix.suffix }}
-      - uses: actions/upload-artifact@v2
-        name: Upload artifacts
+          export _NAME=$(jq ".[\"$GOOS-$GOARCH$GOARM$GOMIPS\"].friendlyName" -r < .github/build/friendly-filenames.json)
+          echo "GOOS: $GOOS, GOARCH: $GOARCH, GOARM: $GOARM, GOMIPS: $GOMIPS, RELEASE_NAME: $_NAME"
+          echo "::set-output name=ASSET_NAME::$_NAME"
+          echo "ASSET_NAME=$_NAME" >> $GITHUB_ENV
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
         with:
-          name: artifacts
-          path: ./artifacts
+          go-version: ^1.16
+ 
+      - name: Build Transfersh
+        run: |
+          mkdir -p build_assets
+          go build -tags netgo -ldflags "-X github.com/dutchcoders/transfer.sh/cmd.Version=${GITHUB_REF##*/} -a -s -w -extldflags '-static'" -o build_assets/transfersh
+    
+      - name: Build Mips softfloat Transfersh
+        if: matrix.goarch == 'mips' || matrix.goarch == 'mipsle'
+        run: |
+          GOMIPS=softfloat go build -tags netgo -ldflags "-X github.com/dutchcoders/transfer.sh/cmd.Version=${GITHUB_REF##*/} -a -s -w -extldflags '-static'" -o build_assets/transfersh_softfloat
 
-  release:
-    runs-on: ubuntu-latest
-    needs: [ build ]
-    steps:
-      - uses: actions/download-artifact@v2
-        name: Download artifacts
+      - name: Rename Windows Transfersh
+        if: matrix.goos == 'windows'
+        run: |
+          cd ./build_assets || exit 1
+          mv transfersh transfersh.exe
+
+      - name: Create ZIP archive
+        shell: bash
+        run: |
+          pushd build_assets || exit 1
+          touch -mt $(date +%Y01010000) *
+          zip -9vr ../Transfersh-$ASSET_NAME.zip .
+          popd || exit 1
+          FILE=./Transfersh-$ASSET_NAME.zip
+          DGST=$FILE.dgst
+          for METHOD in {"md5","sha1","sha256","sha512"}
+          do
+            openssl dgst -$METHOD $FILE | sed 's/([^)]*)//g' >> $DGST
+          done
+
+      - name: Change the name
+        run: |
+          mv build_assets Transfersh-$ASSET_NAME
+
+      - name: Upload files to Artifacts
+        uses: actions/upload-artifact@v2
         with:
-          name: artifacts
-          path: ./artifacts
-      - name: Publish artifacts
-        uses: softprops/action-gh-release@v1
+          name: Transfersh-${{ steps.get_filename.outputs.ASSET_NAME }}
+          path: |
+            ./Transfersh-${{ steps.get_filename.outputs.ASSET_NAME }}/*
+
+      - name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v2
```

Comment:
`svenstaro/upload-release-action@v2` doesn't seem to be actively maintained, I'd stick to `softprops/action-gh-release@v1` if possible